src/export/ExportFilePanel.cpp

```diff
@@ -371,23 +372,36 @@ void ExportFilePanel::ValidateAndFixExt()
 
    wxFileName filename;
    filename.SetFullName(mFullName->GetValue());
+   const auto desiredExt = filename.GetExt().Trim();
 
-   //Remove extra whitespaces
-   auto desiredExt = filename.GetExt().Trim().Trim(false);
-
-   auto it = std::find_if(
-      formatInfo.extensions.begin(),
-      formatInfo.extensions.end(),
-      // if typed extension uses different case (e.g. MP3 instead of mp3)
-      // we'll reset the file extension to one provided by FormatInfo
-      [&](const auto& ext) { return desiredExt.IsSameAs(ext, false); });
+   //See https://github.com/audacity/audacity/issues/5823
+   //check if extension is valid, i.e. does not contain whitespace characters.
+   //Otherwise everything after '.'(if present) is considered to be a part of name.
+   if(wxRegEx{R"(^[^ ]+$)"}.Matches(desiredExt))
+   {
+      auto it = std::find_if(
+         formatInfo.extensions.begin(),
+         formatInfo.extensions.end(),
+         // if typed extension uses different case (e.g. MP3 instead of mp3)
+         // we'll reset the file extension to one provided by FormatInfo
+         [&](const auto& ext) { return desiredExt.IsSameAs(ext, false); });
 
-   if(it == formatInfo.extensions.end())
-      it = formatInfo.extensions.begin();
+      if(it == formatInfo.extensions.end())
+         it = formatInfo.extensions.begin();
 
-   if(!it->empty() && !it->IsSameAs(filename.GetExt()))
+      if(!it->empty() && !it->IsSameAs(filename.GetExt()))
+      {
+         filename.SetExt(*it);
+         mFullName->SetValue(filename.GetFullName());
+      }
+   }
+   else if(!formatInfo.extensions.front().empty())
    {
-      filename.SetExt(*it);
+      auto fullname = filename.GetFullName();
+      if(!fullname.EndsWith("."))
+         fullname.Append(".");
+      fullname.Append(formatInfo.extensions.front());
+      filename.SetFullName(fullname);
       mFullName->SetValue(filename.GetFullName());
```

Comment:
`wxFileName` is not a string, though it can be converted to a string and created from string. Likely the reasoning was that `filename->SetFullName(fullname) == fullname` isn't always the case. But in case when `fullname` originiates from filename it could be indeed overcomplication.